src/App.tsx

```diff
@@ -1,61 +1,110 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { Todo, User } from './react-app-env';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todosWithUser, setTodosWithUser] = useState<Todo[]>(todos);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [selectedUserById, setSelectedUserById] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
```

Comment:
I like your approach here, gj 👍 
It can be improved though. It is a good practice to start with positive - that helps to read the code easier.
```suggestion
  const [isTitleValid, setIsTitleValid] = useState(true);
  const [isUserIdValid, setIsUserIdValid] = useState(true);
```

Reply:
just a suggestion here